### PR TITLE
Claims on token endpoint and Capabilities

### PR DESCRIPTION
**Pull request by jasoncoolmax**
Changes needed in common core is here: https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/271

### CODE REVIEW

File: MSAL/src/public/MSALPublicClientApplication.h [hunk 1]
```diff
@@ -515,6 +520,36 @@
...
+ identifier within the directory itself (e.g. a domain associated
+ to the tenant, such as contoso.onmicrosoft.com, or the GUID
+ representing the TenantID property of the directory)
+ @param  claims                  The claims parameter that needs to be sent to token endpoint. When claims
+ is passed, access token will be skipped and refresh token will be tried.
+ @param  forceRefresh            Ignore any existing access token in the cache and force MSAL to
+ get a new access token from the service.
+ @param  correlationId           UUID to correlate this request with the server
+ @param  completionBlock         The completion block that will be called when the authentication
+ flow completes, or encounters an error.
```
Comment (oldalton) [inline MSAL/src/public/MSALPublicClientApplication.h]: nit: can you please fix alignment of the documentation :)?

File: MSAL/src/requests/MSALRequestParameters.h [hunk 2]
```diff
@@ -69,4 +71,6 @@
 #pragma mark Methods
 - (void)setScopesFromArray:(NSArray<NSString *> *)array;
 - (void)setCloudAuthorityWithCloudHostName:(NSString *)cloudHostName;
+- (void)setClaims:(NSString *)claims NS_UNAVAILABLE;
```
Comment (oldalton) [inline MSAL/src/requests/MSALRequestParameters.h]: can we just remove the property claims instead?

File: MSAL/src/requests/MSALRequestParameters.h [hunk 3]
```diff
@@ -69,4 +71,6 @@
...
+- (BOOL)setClaims:(NSString *)claims error:(NSError **)error;
```
Comment (oldalton) [inline MSAL/src/requests/MSALRequestParameters.h]: Can this live in MSID space instead?

File: MSAL/src/MSALPublicClientApplication.m [hunk 4]
```diff
@@ -572,6 +592,7 @@ - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
     params.account = account;
     params.validateAuthority = _validateAuthority;
     params.sliceParameters = _sliceParameters;
+    params.clientCapabilities = _clientCapabilities;
```
Comment (unpluggedk) [inline MSAL/src/MSALPublicClientApplication.m]: can you also add into the log?

File: MSAL/src/requests/MSALInteractiveRequest.m [hunk 5]
```diff
@@ -145,9 +146,10 @@ - (void)acquireTokenImpl:(MSALCompletionBlock)completionBlock
     config.uid = _parameters.account.homeAccountId.objectId;
     config.utid = _parameters.account.homeAccountId.tenantId;
     config.extraQueryParameters = _parameters.extraQueryParameters;
-    config.claims = _parameters.claims;
     config.sliceParameters = _parameters.sliceParameters;
-
+    NSString *claims = [MSIDClientCapabilitiesUtil msidClaimsParameterFromCapabilities:_parameters.clientCapabilities
+                                                                       developerClaims:_parameters.decodedClaims];
+    if (![NSString msidIsStringNilOrBlank:claims]) config.claims = claims;
```
Comment (unpluggedk) [inline MSAL/src/requests/MSALInteractiveRequest.m]: nit:  parenthesis?

File: MSAL/src/public/MSALPublicClientApplication.h (hunk 1 above)
Comment (jasoncoolmax) [reply MSAL/src/public/MSALPublicClientApplication.h]: fixed.

File: MSAL/src/MSALPublicClientApplication.m (hunk 4 above)
Comment (jasoncoolmax) [reply MSAL/src/MSALPublicClientApplication.m]: fixed :)

File: MSAL/src/requests/MSALInteractiveRequest.m (hunk 5 above)
Comment (jasoncoolmax) [reply MSAL/src/requests/MSALInteractiveRequest.m]: fixed :)

File: MSAL/src/requests/MSALRequestParameters.h (hunk 3 above)
Comment (jasoncoolmax) [reply MSAL/src/requests/MSALRequestParameters.h]: Talked Olga offline, probably we need to have one request parameter class in MSID space such that the duplicate logic could be unified :)

File: MSAL/src/requests/MSALRequestParameters.h (hunk 2 above)
Comment (jasoncoolmax) [reply MSAL/src/requests/MSALRequestParameters.h]: addressed by marking claims property to be readonly :)

**Pull request merged**